components/submodules/submodules.js

```diff
@@ -83,3 +85,23 @@ SubmodulesViewModel.prototype.submoduleLinkClick = function(submodule) {
 SubmodulesViewModel.prototype.submodulePathClick = function(submodule) {
   window.location.href = document.URL + '/' + submodule.path;
 }
+
+SubmodulesViewModel.prototype.submoduleRemove = function(submodule) {
+  var self = this;
+  var diag = components.create('yesnodialog', { title: 'Are you sure?', details: 'This operation cannot be undone with ungit.'});
+  diag.closed.add(function() {
+    if (diag.result()) {
+      self.addProgressBar.start();
+      self.server.del('/submodules', { path: self.repoPath, submodulePath: submodule.path, submoduleName: submodule.name }, function(err, result) {
+        if (err) {
+          console.log(err);
+          return;
+        }
+
+        programEvents.dispatch({ event: 'submodule-added' });
```

Comment:
I think the event should be `submodule-removed`. You can also add it here: https://github.com/codingtwinky/ungit/blob/513/components/submodules/submodules.js#L21